201-front-door-premium-vm-private-link/modules/network.bicep

```diff
@@ -0,0 +1,46 @@
+@description('The location into which the virtual network resources should be deployed.')
+param location string
+
+@description('The IP address prefix (CIDR range) to use when deploying the virtual network.')
+param vnetIPPrefix string
+
+@description('The IP address prefix (CIDR range) to use when deploying the VM subnet within the virtual network.')
+param vmSubnetIPPrefix string
+
+@description('The IP address prefix (CIDR range) to use when deploying the Private Link service environment subnet within the virtual network.')
+param privateLinkServiceSubnetIPPrefix string
+
+var vnetName = 'VNet'
+var vmSubnetName = 'VMs'
+var privateLinkServiceSubnetName = 'PrivateLinkService'
+
+resource vnet 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: vnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetIPPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: vmSubnetName
+        properties: {
+          addressPrefix: vmSubnetIPPrefix
+        }
+      }
+      {
+        name: privateLinkServiceSubnetName
+        properties: {
+          addressPrefix: privateLinkServiceSubnetIPPrefix
+          privateLinkServiceNetworkPolicies: 'Disabled'
+        }
+      }
+    ]
+  }
+}
+
+output vnetName string = vnetName
+output vmSubnetResourceId string = resourceId('Microsoft.Network/virtualNetworks/subnets', vnetName, vmSubnetName)

```

Comment:
here

vnet.properties.subnets[0].id

should work

Reply:
Agreed - but I'm generally not a fan of index-based properties, because it's brittle to changes in the ordering of the subnets. Referring to it by name instead of ordinal index seems more stable. What do you think?

Reply:
Very good point - does bicep allow this?

vnet.properties.subnets[vmSubnetName].id

ideally we want to make it easier to get to resourceIds in bicep (so obviate the resourceId function) but your concern trumps that imo

Reply:
No it doesn't, unfortunately. I opened a Bicep issue for this: https://github.com/Azure/bicep/issues/1853

I'll close this for now, but if that gets fixed I'll adjust the template accordingly.